web/src/ResourceInfoKeyboardShortcuts.test.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react"
+import { mount } from "enzyme"
+import ResourceInfoKeyboardShortcuts from "./ResourceInfoKeyboardShortcuts"
+import { fireEvent } from "@testing-library/dom"
+
+function numKeyCode(num: number): number {
+  return num + 48
+}
+
+type Link = Proto.webviewLink
+
+let openedSnapshotModal = false
+let endpointUrl = ""
+const shortcuts = (endpoints: Link[]) => {
+  endpointUrl = ""
+  openedSnapshotModal = false
+  mount(
+    <ResourceInfoKeyboardShortcuts
+      endpoints={endpoints}
+      showSnapshotButton={true}
+      openEndpointUrl={url => (endpointUrl = url)}
+      openSnapshotModal={() => (openedSnapshotModal = true)}
+    />
+  )
+}
+
+it("zero endpoint urls", () => {
+  shortcuts([])
+  fireEvent.keyDown(document.body, { keyCode: numKeyCode(1), shiftKey: true })
+  expect(endpointUrl).toEqual("")
+})
+it("two endpoint urls trigger first", () => {
+  shortcuts([
+    { url: "https://tilt.dev:4000" },
+    { url: "https://tilt.dev:4001" },
+  ])
+  fireEvent.keyDown(document.body, { keyCode: numKeyCode(1), shiftKey: true })
+  expect(endpointUrl).toEqual("https://tilt.dev:4000")
+})
+it("two endpoint urls trigger second", () => {
+  shortcuts([
+    { url: "https://tilt.dev:4000" },
+    { url: "https://tilt.dev:4001" },
+  ])
+  fireEvent.keyDown(document.body, { keyCode: numKeyCode(2), shiftKey: true })
+  expect(endpointUrl).toEqual("https://tilt.dev:4001")
+})
+it("handles snapshot shortcut", () => {
```

Comment:
nbd but consider a test for:`showSnapshotButton` is false, `s` does nothing

Reply:
done